### ISSUE TITLE
joinString() fails if first element of list is an empty string

### ISSUE THREAD

**Issue by end2endzone** [end2endzone/RapidAssist#11]
The function does not add a separator at the beginning of the returned string if the first element of list is an empty string.

The following unit test can be used to reproduce the issue:
```cpp
      static const std::string EXPECTED = ".Aa.Bb.Cc";
      static const char * SEPARATOR = ".";
      StringVector list = splitString(EXPECTED, SEPARATOR);
      std::string joinStr = joinString(list, SEPARATOR);
      ASSERT_EQ(EXPECTED, joinStr);
```

**Comment by end2endzone** [comment]
This bug affects version 0.3.x, 0.4.x

**Comment by end2endzone** [comment]
Issue #6, #7 and #11 resolved in 113da57c70484d78d2e1880037020ea98a6d64ed
